org.eclipse.xtext.builder.standalone/src/org/eclipse/xtext/builder/standalone/incremental/ResourceURICollector.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) 2015, 2019 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.xtext.builder.standalone.incremental;
+
+import static java.util.stream.Collectors.*;
+import static java.util.stream.StreamSupport.*;
+
+import java.io.File;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+import java.util.zip.ZipException;
+
+import org.apache.log4j.Logger;
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.plugin.EcorePlugin;
+import org.eclipse.xtext.mwe.PathTraverser;
+import org.eclipse.xtext.xbase.lib.IterableExtensions;
+
+import com.google.common.collect.Multimap;
+
+/**
+ * @author Jan Koehnlein - Initial contribution and API
+ * @since 2.9
+ */
+public class ResourceURICollector {
+	private static final Logger LOG = Logger.getLogger(ResourceURICollector.class);
+
+	public Set<URI> collectAllResources(Iterable<URI> roots, Set<String> fileExtensions) {
+		if (LOG.isInfoEnabled()) {
+			LOG.info("Collecting source models.");
+		}
+		long startedAt = System.currentTimeMillis();
+		Set<URI> result = collectResources(roots, fileExtensions);
+		if (LOG.isDebugEnabled()) {
+			LOG.debug("Finished collecting source models. Took: " + (System.currentTimeMillis() - startedAt) + " ms.");
+		}
+		return result;
+	}
+
+	protected Set<URI> collectResources(Iterable<URI> roots, Set<String> fileExtensions) {
+		Set<String> extensions = new HashSet<>(fileExtensions);
+		extensions.add("java");
+
+		List<String> paths = stream(roots.spliterator(), false).map(uri -> uri.toFileString()).collect(toList());
```

Comment:
Can't we use `map(URI::toFileString)` here?

Reply:
yes; I'll also add a test case